### DISCUSSION TITLE
How to resolve link in sidebar or nav for non-markdown file

### DISCUSSION THREAD

**Discussion by liuliangsir** [vuejs/vitepress#1633] [Q&A] [answered]
As the title above says, I would like to resolve link in sidebar or nav for non-markdown file like json file. Vitepress now adds `.html` extension for all the non-markdown files. It seems to be a bug since that behavior could result in a bunch of broken urls when navigating from sidebar or nav. Is there any workaround to solve the issue above?

**Comment by liuliangsir** [comment, marked as answer]
@brc-dd cc

**Reply by liuliangsir** [reply]
@erondpowell cc

**Reply by brc-dd** [reply]
Use absolute link for that. `https://foo.bar/baz.json`

**Comment by Zhengqbbb** [comment]
Your mean is sidebar and nav bar configure setting use `json` file?
Or other ? Please provide u expected
